src/Features/CSharp/Portable/ChangeSignature/CSharpChangeSignatureService.cs

```diff
@@ -442,31 +450,100 @@ private SyntaxNode GetNodeContainingTargetNode(SyntaxNode matchingNode)
+                    ConstructorInitializerSyntax constructorInitializer => constructorInitializer.ArgumentList,
+                    ElementAccessExpressionSyntax elementAccess => elementAccess.ArgumentList,
+                    _ => throw new ArgumentException("Unexpected SyntaxNode", nameof(node))
+                };
+
+                argumentCount = argumentList.Arguments.Count;
+                lastArgumentHasNameColon = argumentList.Arguments.LastOrDefault()?.NameColon != null;
+                lastArgumentExpression = argumentList.Arguments.LastOrDefault()?.Expression;
+            }
+
+
+            if (symbolInfo.Symbol is IMethodSymbol methodSymbol && methodSymbol.Parameters.LastOrDefault()?.IsParams == true)
+            {
+                if (argumentCount > ((IMethodSymbol)symbolInfo.Symbol).Parameters.Length)
+                {
+                    return true;
+                }
+                else if (argumentCount == ((IMethodSymbol)symbolInfo.Symbol).Parameters.Length)
+                {
+                    if (lastArgumentHasNameColon)
+                    {
+                        return true;
+                    }
+                    else
+                    {
+                        var fromType = semanticModel.GetTypeInfo(lastArgumentExpression, cancellationToken);
+                        var toType = ((IMethodSymbol)symbolInfo.Symbol).Parameters.Last().Type;
+                        return !semanticModel.Compilation.HasImplicitConversion(fromType.Type, toType);
+                    }
+                }
+                else
+                {
+                    return false;
+                }
+            }
+
+            return false;
+        }
+
         private static ParameterSyntax CreateNewParameterSyntax(AddedParameter addedParameter)
             => CreateNewParameterSyntax(addedParameter, skipParameterType: false);
 
         private static ParameterSyntax CreateNewParameterSyntax(AddedParameter addedParameter, bool skipParameterType)
-            => SyntaxFactory.Parameter(
-                attributeLists: SyntaxFactory.List<AttributeListSyntax>(),
-                modifiers: SyntaxFactory.TokenList(),
+        {
+            var equalsValueClause = addedParameter.HasDefaultValue
+                ? EqualsValueClause(ParseExpression(addedParameter.DefaultValue))
+                : null;
+
+            return Parameter(
+                attributeLists: default,
+                modifiers: default,
                 type: skipParameterType
                     ? null
-                    : addedParameter.Type.GenerateTypeSyntax(allowVar: false)
-                        .WithTrailingTrivia(SyntaxFactory.ElasticSpace),
-                SyntaxFactory.Identifier(addedParameter.ParameterName),
-                @default: null);
+                    : addedParameter.Type.GenerateTypeSyntax()
+                        .WithTrailingTrivia(ElasticSpace),
```

Comment:
adding the trailing space seems like it should be unnecessary.

Reply:
It was.